docs/architecture.md

```diff
@@ -0,0 +1,82 @@
+# ODH Dashboard Architecture
+
+## Overview
+
+![Overview](./meta/arch-overview.png)
+
+> **OpenShift OAuth Proxy** is not part of the Dashboard (see [Dashboard Deployment `containers`](https://github.com/opendatahub-io/odh-dashboard/blob/main/manifests/base/deployment.yaml)) but useful to understand its role. See [OpenShift OAuth Proxy repo](https://github.com/openshift/oauth-proxy) for more information. It is typically in the containers of all services we talk to from the proxy API, as well as the Dashboard itself when the client calls the NodeJS server (in the pod).
+
+The main focus point here is that there are 5 kinds of calls from the client. These are split into two different types. The difference in these two types is one takes your request and does it on your behalf (via the service account), and the other does it as you (through direct calls with your bearer token).
+
+### Custom Business Logic Backend
+
+> **Note**: this functionality is deprecated
+
+These are made on your behalf. There is a Dashboard service account (`1`) that makes these calls on your behalf to the k8s server to create, read, update, and delete resources. This is usually done with impunity (`2`).
+
+- Framework call
+    - These are calls that get us your username, your allowed/admin status, the feature flags (OdhDashboardConfig), and other various infrastructure based items
+- Jupyter / Admin call
```

Comment:
The doc looks great, the only thing I miss (not sure if it's worth mentioning) is that the backend logic has it's own user/admin rbac (that ends up using the dashboard SA), it's already documented [here](https://github.com/opendatahub-io/odh-dashboard/blob/6558a9fa23e8aacb26a04d99f1d236d1172e8c22/docs/admin_dashboard.md) but it might be relevant  in the architecture overview that we override OpenShift's RBAC

Reply:
Oh, sorry, the custom Fastify Route wrapper stuff ... yeah... not sure I want to mention it tbh. But it is part of the architecture 🤔 Maybe I can make a note -- I didn't want to expose too much of the nitty gritty as there are things about the code that are there but just not talked about. It is unique, maybe I can add a line of detail to help that along.

Reply:
I went to add it and it's covered in a vague but satisfactory (imo) way as the `2` note at the bottom of the section. I'm going to resolve this as-is and we can swing back to worry about further details as we progress forward.